### PR TITLE
Feature/native toast

### PR DESCRIPTION
**Pull request by fuatguzel**

### CODE REVIEW

File: scripts/components/LviCartItem.ts [hunk 1]
```diff
@@ -19,7 +21,7 @@ export default class LviCartItem extends LviCartItemDesign {
         this.btnCartMinus.on(Button.Events.Press, () => {
             this._valueMinus && this._valueMinus();
         });
-        this.lblCloseIcon.on(Label.Events.Touch, () => {
+        this.btnClose.on(Button.Events.Press, () => {
             this._removeValue && this._removeValue();
```
Comment (berkbaski) [inline scripts/components/LviCartItem.ts]: We can use null or undefined check like this

```ts
this._removeValue?.();
```

This code will not execute if `this._removeValue` is undefined.

File: scripts/components/LviCartItem.ts [hunk 2]
```diff
@@ -49,10 +51,34 @@ export default class LviCartItem extends LviCartItemDesign {
         });
     }
     get productPrice(): string {
-        return this.lblProductPrice.text;
+        return this.tvPrice.text;
     }
     set productPrice(value: string) {
-        this.lblProductPrice.text = value;
+        const discountExists = !!this.tvDiscount.text;
+        let attributeString = new AttributedString({
```
Comment (berkbaski) [inline scripts/components/LviCartItem.ts]: We should use const if the variable will not change in the next lines.

File: scripts/components/LviCartItem.ts [hunk 3]
```diff
@@ -49,10 +51,34 @@ export default class LviCartItem extends LviCartItemDesign {
...
+            foregroundColor: themeService.getNativeStyle(discountExists ? '.product-price.discount' : '.product-price.nodiscount').textColor
+        });
+        this.tvPrice.scrollEnabled = false;
+        this.tvPrice.attributedText = [attributeString];
+    }
+    get productDiscount(): string {
+        return this.tvDiscount.text;
+    }
+    set productDiscount(value: string) {
+        let attributeString = new AttributedString({
```
Comment (berkbaski) [inline scripts/components/LviCartItem.ts]: We should use const if the variable will not change in the next lines.

File: scripts/components/LviCartItem.ts [hunk 4]
```diff
@@ -49,10 +51,34 @@ export default class LviCartItem extends LviCartItemDesign {
...
+            string: value || '',
+            font: themeService.getNativeStyle('.product-price.nodiscount').font,
```
Comment (berkbaski) [inline scripts/components/LviCartItem.ts]: Getting theme values should top on the page class.

File: scripts/components/LviCartItem.ts [hunk 5]
```diff
@@ -49,10 +51,34 @@ export default class LviCartItem extends LviCartItemDesign {
...
+            foregroundColor: themeService.getNativeStyle('.product-price').textColor
```
Comment (berkbaski) [inline scripts/components/LviCartItem.ts]: Getting theme values should top on the page class.

File: scripts/components/LviFavorites.ts [hunk 6]
```diff
@@ -21,11 +22,35 @@ export default class LviFavorites extends LviFavoritesDesign {
...
         this.lblFavoriteItemTitle.text = value;
     }
-    get itemPrice(): any {
-        return this.lblFavoriteItemPrice.text;
+    get itemPrice(): string {
+        return this.tvPrice.text;
+    }
+    set itemPrice(value: string) {
+        const discountExists = !!this.tvDiscount.text;
+        let attributeString = new AttributedString({
```
Comment (berkbaski) [inline scripts/components/LviFavorites.ts]: We should use const if the variable will not change in the next lines.

File: scripts/components/LviFavorites.ts [hunk 7]
```diff
@@ -21,11 +22,35 @@ export default class LviFavorites extends LviFavoritesDesign {
...
+        this.tvPrice.scrollEnabled = false;
+        this.tvPrice.attributedText = [attributeString];
+    }
+    get itemDiscount(): string {
+        return this.tvDiscount.text;
     }
-    set itemPrice(value: any) {
-        this.lblFavoriteItemPrice.text = value;
+    set itemDiscount(value: string) {
+        let attributeString = new AttributedString({
```
Comment (berkbaski) [inline scripts/components/LviFavorites.ts]: We should use const if the variable will not change in the next lines.

File: scripts/pages/pgCart.ts [hunk 8]
```diff
@@ -70,7 +70,8 @@ export default class PgCart extends withDismissAndBackButton(PgCartDesign) {
                         productName: cart.name,
                         productInfo: cart.shortDescription,
                         productImage: cart.images ? getProductImageUrl(cart.images[0]) : null,
-                        productPrice: cart.discountPrice != undefined ? `$${cart?.discountPrice?.toFixed(2)}` : `$${cart.price.toFixed(2)}`,
+                        productDiscount: cart.discountPrice != undefined ? `$${cart?.discountPrice?.toFixed(2)}` : '',
```
Comment (berkbaski) [inline scripts/pages/pgCart.ts]: Maybe we should create a function for money formatter. And then we would use the function when we need display money things...

File: scripts/store/main/reducer.ts [hunk 9]
```diff
@@ -57,8 +57,13 @@ export default function (state = initialState, action: ActionTypes): SessionStat
             break;
         }
         case Constants.ADD_TO_BASKET: {
-            showToastDialog();
-            action.payload.count == 1 ? callVibrate(VIBRATION_ADD_TIME) : callVibrate(VIBRATION_REMOVE_TIME);
+            if (action.payload.count == -1) {
```
Comment (berkbaski) [inline scripts/store/main/reducer.ts]: We should write three equals for exact results :)

File: scripts/store/main/reducer.ts [hunk 10]
```diff
@@ -57,8 +57,13 @@ export default function (state = initialState, action: ActionTypes): SessionStat
...
+                removeFromBasketToast();
+                callVibrate(VIBRATION_REMOVE_TIME);
+            } else {
+                addToBasketToast();
+                callVibrate(VIBRATION_ADD_TIME);
+            }
             if (newState.basket.some((pId) => pId._id === action.payload.product._id)) {
                 let updatedData = newState.basket.map((basketItem) =>
```
Comment (berkbaski) [inline scripts/store/main/reducer.ts]: We should use const if the variable will not change in the next lines.

File: scripts/pages/pgFavorites.ts [hunk 11]
```diff
@@ -161,10 +161,8 @@ export default class PgFavorites extends withDismissAndBackButton(PgFavoritesDes
                             itemTitle: favouritedItem.name,
                             itemDesc: favouritedItem.shortDescription,
                             itemImage: favouritedItem.images ? getProductImageUrl(favouritedItem.images[0]) : null,
-                            itemPrice:
-                                favouritedItem.discountPrice != undefined
-                                    ? `$${favouritedItem?.discountPrice?.toFixed(2)}`
-                                    : `$${favouritedItem.price.toFixed(2)}`,
+                            itemDiscount: favouritedItem.discountPrice != undefined ? `$${favouritedItem?.discountPrice?.toFixed(2)}` : '',
+                            itemPrice: `$${favouritedItem.price.toFixed(2)}`,
```
Comment (berkbaski) [inline scripts/pages/pgFavorites.ts]: Maybe we should create a function for money formatter. And then we would use the function when we need display money things...

File: scripts/lib/toast.ts [hunk 12]
```diff
@@ -1,54 +1,29 @@
...
-    themeService.addGlobalComponent(component as any, 'flToast');
-    (component as StyleContextComponentType<FlexLayout>).dispatch({
-        type: 'pushClassNames',
-        classNames: '.flToast'
+export const addToBasketToast = () => {
+    System.OS === System.OSType.ANDROID ? console.info('AND') : console.info('IOS');
+    const myToastMessage = new Toast({
+        message: global.lang.addedToBasket,
+        bottomOffset: System.OS === System.OSType.ANDROID ? bottomOffsetAnd : bottomOffsetIOS,
+        backgroundColor: themeService.getNativeStyle('.flProductItemButtonsWrapper-btnMinus.main').textColor,
```
Comment (berkbaski) [inline scripts/lib/toast.ts]: Getting theme values should top on the page class.

File: scripts/lib/toast.ts [hunk 13]
```diff
@@ -1,54 +1,29 @@
...
-        }
-    }) as StyleContextComponentType<Dialog>;
-    const component = new FlToast()
-    component.title = global.lang.addToBasket;
-    themeService.addGlobalComponent(component as any, 'flToast');
-    (component as StyleContextComponentType<FlexLayout>).dispatch({
-        type: 'pushClassNames',
-        classNames: '.flToast'
+export const addToBasketToast = () => {
+    System.OS === System.OSType.ANDROID ? console.info('AND') : console.info('IOS');
```
Comment (berkbaski) [inline scripts/lib/toast.ts]: We don't need any logs :)

File: scripts/lib/toast.ts [hunk 14]
```diff
@@ -1,54 +1,29 @@
...
-            setTimeout(() => toastDialog.hide(), timeout);
-        } else {
-            toastDialog.hide();
-        }
-    }
+export const removeFromBasketToast = () => {
+    const myToastMessage = new Toast({
+        message: global.lang.removedFromBasket,
+        bottomOffset: System.OS === System.OSType.ANDROID ? bottomOffsetAnd : bottomOffsetIOS,
+        backgroundColor: themeService.getNativeStyle('.flProductItemButtonsWrapper-btnMinus.danger').textColor,
```
Comment (berkbaski) [inline scripts/lib/toast.ts]: Getting theme values should top on the page class.

**Review by berkbaski — COMMENTED** [review]
I added some change requests. Could you fix these? @fuatguzel

File: scripts/components/LviCartItem.ts (hunk 1 above)
Comment (fuatguzel) [reply scripts/components/LviCartItem.ts]: DONE.

File: scripts/components/LviCartItem.ts (hunk 2 above)
Comment (fuatguzel) [reply scripts/components/LviCartItem.ts]: DONE.

File: scripts/components/LviCartItem.ts (hunk 3 above)
Comment (fuatguzel) [reply scripts/components/LviCartItem.ts]: DONE.

File: scripts/components/LviCartItem.ts (hunk 4 above)
Comment (fuatguzel) [reply scripts/components/LviCartItem.ts]: DONE.

File: scripts/components/LviCartItem.ts (hunk 5 above)
Comment (fuatguzel) [reply scripts/components/LviCartItem.ts]: DONE.

File: scripts/components/LviFavorites.ts (hunk 6 above)
Comment (fuatguzel) [reply scripts/components/LviFavorites.ts]: DONE.

File: scripts/components/LviFavorites.ts (hunk 7 above)
Comment (fuatguzel) [reply scripts/components/LviFavorites.ts]: DONE.

File: scripts/store/main/reducer.ts (hunk 9 above)
Comment (fuatguzel) [reply scripts/store/main/reducer.ts]: DONE.

File: scripts/store/main/reducer.ts (hunk 10 above)
Comment (fuatguzel) [reply scripts/store/main/reducer.ts]: DONE.

File: scripts/lib/toast.ts (hunk 12 above)
Comment (fuatguzel) [reply scripts/lib/toast.ts]: DONE.

File: scripts/lib/toast.ts (hunk 13 above)
Comment (fuatguzel) [reply scripts/lib/toast.ts]: DONE.

File: scripts/lib/toast.ts (hunk 14 above)
Comment (fuatguzel) [reply scripts/lib/toast.ts]: DONE.

File: scripts/pages/pgCart.ts (hunk 8 above)
Comment (fuatguzel) [reply scripts/pages/pgCart.ts]: DONE.

File: scripts/pages/pgFavorites.ts (hunk 11 above)
Comment (fuatguzel) [reply scripts/pages/pgFavorites.ts]: DONE.

File: scripts/pages/pgCart.ts [hunk 15]
```diff
@@ -70,8 +71,8 @@ export default class PgCart extends withDismissAndBackButton(PgCartDesign) {
                         productName: cart.name,
                         productInfo: cart.shortDescription,
                         productImage: cart.images ? getProductImageUrl(cart.images[0]) : null,
-                        productDiscount: cart.discountPrice != undefined ? `$${cart?.discountPrice?.toFixed(2)}` : '',
-                        productPrice: `$${cart.price.toFixed(2)}`,
+                        productDiscount: cart.discountPrice != undefined ? moneyFormatter(cart?.discountPrice) : '',
```
Comment (berkbaski) [inline scripts/pages/pgCart.ts:74]: Maybe we can use like this

```ts
productDiscount: moneyFormatter(cart?.discountPrice) || '',
```

**Comment by berkbaski** [comment]
LGTM! 💯

**Pull request merged**